lib/rubocop/rspec/top_level_group.rb

```diff
@@ -38,7 +38,9 @@ def top_level_group?(node)
       end
 
       def top_level_nodes(node)
-        if node.begin_type?
+        if node.nil?
+          []
```

Comment:
I remember we were [discussing this with Maxim](https://github.com/rubocop-hq/rubocop-rspec/pull/977#discussion_r462301691) and settled on
```ruby
return unless root_node
```

Do you think both checks are necessary?

Reply:
As you mention, we already have `return unless root_node` inside `#on_new_investigation`. The problem happens when the root node is an empty module or class:

```rb
elsif node.module_type? || node.class_type?
  top_level_nodes(node.body)
```

Reply:
I mean, we could move the check into the `elsif` body:
```rb
elsif node.module_type? || node.class_type?
  node.body.nil? ? [] : top_level_nodes(node.body)
```
or
```rb
elsif node.module_type? || node.class_type?
  if node.body
    top_level_nodes(node.body)
  else
    []
  end
```